tests/ZfrOAuth2Test/Server/Grant/AuthorizationGrantTest.php

```diff
@@ -244,7 +255,7 @@ public function testCanCreateTokenResponse($hasRefreshGrant)
 
         $response = $this->grant->createTokenResponse($request, new Client(), $owner);
 
-        $body = json_decode($response->getContent(), true);
+        $body = json_decode($response->getBody(), true);
```

Comment:
Try calling $stream->rewind() either directly after writing to it, or
directly before grabbing the contents. getContents() is expected to return
contents from the current pointer position, unless an offset is provided.
On May 24, 2015 11:04 AM, "Michaël Gallego" <notifications@github.com>
wrote:

> In tests/ZfrOAuth2Test/Server/Grant/AuthorizationGrantTest.php
> <https://github.com/zf-fr/zfr-oauth2-server/pull/31#discussion_r30953467>:
>
> > @@ -244,7 +255,7 @@ public function testCanCreateTokenResponse($hasRefreshGrant)
> >
> >          $response = $this->grant->createTokenResponse($request, new Client(), $owner);
> >
> > -        $body = json_decode($response->getContent(), true);
> > +        $body = json_decode($response->getBody(), true);
>
> Hi @weierophinney <https://github.com/weierophinney> . It does not seem
> to work. Here is how I create the response:
>
>  $stream = new Stream('php://temp');        $stream->write(json_encode(['error' => $exception->getCode(), 'error_description' => $exception->getMessage()]));        return new Response($stream, 400);
>
> then in my tests:
>
> $body     = json_decode($response->getBody(), true);
>
> $body is NULL. I've tried getBody()->getContents(), but I retrieve an
> empty string. How should it work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zf-fr/zfr-oauth2-server/pull/31/files#r30953467>.
>